R/FilteredData.R

```diff
@@ -315,29 +352,51 @@ FilteredData <- R6::R6Class( # nolint
     },
 
     #' @description
-    #' Adds a `TealDataset` object to this `FilteredData`
+    #' Adds a dataset to this `FilteredData`
     #'
-    #' Adds a dataset and preserve all attributes attached to this object.
     #' Technically `set_dataset` created `FilteredDataset` which keeps
     #' `dataset` for filtering purpose.
     #'
-    #' @param dataset (`TealDataset` or `TealDatasetConnector`)\cr
-    #'   the object containing data and attributes.
-    #' @return (`self`) invisibly this `FilteredTealData`
-    set_dataset = function(dataset) {
-      stopifnot(is(dataset, "TealDataset") || is(dataset, "TealDatasetConnector"))
-      dataname <- teal.data::get_dataname(dataset)
+    #' @param dataset_args (`list`)\cr
+    #'   containing the arguments except (`dataname`)
+    #'   needed by `init_filtered_dataset`
+    #' @param dataname (`string`)\cr
+    #'   the name of the `dataset` to be added to this object
+    #' @return (`self`) invisibly this `FilteredData`
+    set_dataset = function(dataset_args, dataname) {
```

Comment:
I'd send it to private field now

Reply:
so I was thinking about this - could there be a situation where you've created the filtereddata and then want to add more into it? And not have to recreate the whole object - if so this and join_keys would be public otherwise yup can move to private

Reply:
though if I make it private I don't need to test it 😂

Reply:
Good point, let's keep it as is.